src/nvim/os/input.c

```diff
@@ -82,9 +82,11 @@ static void cursorhold_event(void **argv)
 
 static void create_cursorhold_event(void)
 {
-  // If the queue had any items, this function should not have been
-  // called(inbuf_poll would return kInputAvail)
-  assert(queue_empty(loop.events));
+  // If events are enabled and the queue has any items, this function should not
+  // have been called(inbuf_poll would return kInputAvail)
+  // TODO(tarruda): Cursorhold should be implemented as a timer set during the
+  // `state_check` callback for the states where it can be triggered.
+  assert(!events_enabled || queue_empty(loop.events));
```

Comment:
Do we still need this `assert()`? If events are enabled we already know that `loop.events` is empty. Or could the call `ui_flush()` in `state_enter()` fill the queue before `os_inchar()` is called?